app/questionnaire/router.py

```diff
@@ -398,7 +398,7 @@ def _is_section_enabled(self, section: Mapping) -> bool:
             metadata=self._metadata,
             response_metadata=self._response_metadata,
             progress_store=self._progress_store,
-            location=None,
+            location=Location(section_id=section["id"]),
```

Comment:
Since you use `section["id"]` twice you could just assign it a var similar to how `enabled` is done on line 388.

Reply:
Updated this 👍